internal/controllers/synthetic_devices_controller_test.go

```diff
@@ -209,6 +214,16 @@ func (s *VirtualDevicesControllerTestSuite) TestSignVirtualDeviceMintingPayload(
 	udID := ksuid.New().String()
 	_ = test.SetupCreateVehicleNFTForMiddleware(s.T(), addr, testUserID, udID, 57, s.pdb)
 
+	s.virtDeviceSigSvc.EXPECT().SignHash(gomock.Any(), gomock.Any(), gomock.Any())
+
+	var kb []byte
+	mockProducer.ExpectSendMessageWithCheckerFunctionAndSucceed(func(val []byte) error {
+		kb = val
+		return nil
+	})
+
+	log.Println(string(kb))
```

Comment:
Let's delete this, but check some stuff:

* It's a CloudEvent of the right type
* It's [targeted at the right address](https://github.com/DIMO-Network/devices-api/blob/2aaf10bf377c3b3fbf0edd36032111cf6433b4b9/internal/services/registry/client.go#L34)
* If we [decode the calldata](https://github.com/DIMO-Network/devices-api/blob/2aaf10bf377c3b3fbf0edd36032111cf6433b4b9/internal/controllers/user_integrations_controller_test.go#L881), it's a call to the right function, with the right inputs. You'll probably have to play with this `.Unpack` thing.

I think it's fair to say that the abstraction of `registry.Client` is not great. That's my fault.